### PR TITLE
cpdp-521: allowed get bpr and subscription for trusts

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala [hunk 1]
```diff
@@ -43,20 +44,56 @@ class SubscriptionConnectorImpl @Inject()(http: HttpClient, val config: Services
...
+
+object SubscriptionConnectorImpl {
+
+  sealed trait TypeOfPerson { val value: Int }
+
+  object TypeOfPerson {
+
+    final case object Individual extends TypeOfPerson {
+      val value = 1
+    }
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala]: where do you use this value field?

File: app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala [hunk 2]
```diff
@@ -43,20 +44,56 @@ class SubscriptionConnectorImpl @Inject()(http: HttpClient, val config: Services
...
+  }
+
+}
+
+object SubscriptionConnectorImpl {
+
+  sealed trait TypeOfPerson { val value: Int }
+
+  object TypeOfPerson {
+
```
Comment (co-category) [inline app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala]: I think a better name for this is AffinityType

File: test/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnectorImplSpec.scala [hunk 3]
```diff
@@ -57,11 +58,85 @@ class SubscriptionConnectorImplSpec extends WordSpec with Matchers with MockFact
...
+        "email",
+        UkAddress("line1", None, None, None, "postcode"),
+        "sap"
+      )
+
+      val expectedRequest = Json.parse(
+        """
+          |{
+          |  "typeOfPerson" : "1",
+          |  "contactName"  : "name surname",
```
Comment (co-category) [inline test/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnectorImplSpec.scala]: sigh - DES data modelling at it's finest.

Comment (AndyHWChung) [reply test/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnectorImplSpec.scala]: this is subject to change - perhaps we can get a better model when we negotiate the specs with them if we ask nicely enough

File: app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala (hunk 2 above)
Comment (co-category) [reply app/uk/gov/hmrc/cgtpropertydisposals/connectors/SubscriptionConnector.scala]: OK, this probably makes sense to keep it as TypeOfPerson if DES are expecting that field name in the JSON request.

**Pull request merged**